ddtrace/internal/_encoding.pyx

```diff
@@ -569,12 +569,19 @@ cdef class MsgpackEncoderV03(MsgpackEncoderBase):
 
         for link in span_links:
             # SpanLink.to_dict() returns all serializable span link fields
+            # v0.4 encoding is disabled by default. SpanLinks.to_dict() is optimizied for the v0.5 format.
             d = link.to_dict()
             # Encode 128 bit trace ids usings two 64bit integers
-            d["trace_id_high"] = int(d["trace_id"][:16], 16)
+            tid = int(d["trace_id"][:16], 16)
+            if tid > 0:
```

Comment:
what'll happen if if `tid <0`?

Reply:
Then we don't need to set trace_id_high.  This is an optional field in the payload